pushbullet

```diff
@@ -105,6 +106,22 @@ delete)
 			exit 1
 		fi
 		;;
+	old)
+		if [ "$3" == "" ]; then	printUsage; fi
+		echo "deleting all pushes except the last $3"
+		number=$(($3+1))
+		allpushes=$(curl -s "$API_URL/pushes?active=true" -u "$PB_API_KEY": | "$PROGDIR"/JSON.sh -l)
+		activepushes=$(echo "$allpushes" | egrep "\[\"pushes\",[0-9]+,\"active\"\].*true"|while read line; do echo "$line"|cut -f 2 -d ,; done | tail -n "+$number")
+		for id in $activepushes; do
+			iden=$(echo "$allpushes" | grep "^\[\"pushes\",$id,\"iden\"\]"|cut -f 2 | cut -d'"' -f 2)
+			echo "Deleting \"$iden\""
+	 		curlres=$(curl -s "$API_URL/pushes/$iden" -u "$PB_API_KEY": -X DELETE)
+			if [ ! "$curlres" == '{}' ]; then
+				echo "something went wrong"
+				exit 1
+			fi
+		done
+		;;
 	*)
 		echo "deleting only $2"
```

Comment:
to make more sense of the "delete except $number" then the "only" should be dropped from here.